### DISCUSSION TITLE
Default tls_config client settings for servicemonitors/podmonitors

### DISCUSSION THREAD

**Discussion by joke** [grafana-cold-storage/agent#6136] [Q&A] [answered]
Hi,

I'm using istio with strict mTLS settings. Consequently the grafana agent needs to scrape the prometheus endpoints with special tls settings as described [here](https://istio.io/latest/docs/ops/integrations/prometheus/#tls-settings).

The `prometheus.scrape` components allows specifying a [`tls_config`](https://grafana.com/docs/agent/latest/flow/reference/components/prometheus.scrape/#tls_config-block) block which can be used for this.

Unfortunately the neither [`servicemonitors`](https://grafana.com/docs/agent/latest/flow/reference/components/prometheus.operator.servicemonitors/) nor [`podmonitors`](https://grafana.com/docs/agent/latest/flow/reference/components/prometheus.operator.podmonitors/) support this configuration. The [`client->tls_config`](https://grafana.com/docs/agent/latest/flow/reference/components/prometheus.operator.servicemonitors/#tls_config-block) is used for accessing the kubernetes api and the [`scrape`](https://grafana.com/docs/agent/latest/flow/reference/components/prometheus.operator.servicemonitors/#scrape-block) block does not include a configurations options.

It would be possible to include all the tls configuration in each monitor specification but in addition to being quite repetitive certificate would need to be referenced as secrets instead of files.

Is there any possibility to specify an agent wide http client configuration or an other option to pass a default tls configuration to the scraping process?

Thanks in advance.

**Comment by tpaschalis** [comment]
Hey, thanks for bringing this to our attention 🙌

This is a good usecase for an enhancement, so I'd like to move this to an issue so we can properly keep track of it.

**Comment by tpaschalis** [comment, marked as answer]
If that's OK, I'll close out this discussion and we can continue on https://github.com/grafana/agent/issues/6239
